src/mpl2/src/object.cpp

```diff
@@ -1033,7 +1033,7 @@ void HardMacro::updateDb(float pitch_x, float pitch_y, odb::dbBlock* block)
 
   inst_->setOrigin(snap_origin_x, snap_origin_y);
   inst_->setOrient(orientation_);
-  inst_->setPlacementStatus(odb::dbPlacementStatus::LOCKED);
+  inst_->setPlacementStatus(odb::dbPlacementStatus::PLACED);
```

Comment:
Macros should be locked as we don't want global placement to move them.

Reply:
This is where mpl2 is updating db with the orientation that we are correcting later with the new pass. If they're locked here, we won't be able to adjust the orientation. 
https://github.com/AcKoucher/OpenROAD/blob/82f3cabf7abce001fd8ae8672802a517f2b2fb3f/src/mpl2/src/hier_rtlmp.cpp#L638-L644

The new pass now is responsible for locking the macros:

https://github.com/AcKoucher/OpenROAD/blob/82f3cabf7abce001fd8ae8672802a517f2b2fb3f/src/mpl2/src/hier_rtlmp.cpp#L6033-L6035